foyer/validator.py

```diff
@@ -154,9 +154,13 @@ def validate_smarts(self):
                             None, entry.sourceline)
                         errors.append(undefined)
         raise_collected(errors)
-        if missing_smarts:
+        if missing_smarts and debug:
             warn("The following atom types do not have smarts definitions: {}".format(
                 ', '.join(missing_smarts)), ValidationWarning)
+       	if missing_smarts and not debug:
+       	    warn("There are {} atom types that are missing a smarts definition. "
+                     "To view the missing atom types, re-run with debug=True when "
+                     "applying the forcefield.".format(len(missing_smarts)), ValidationWarning)
```

Comment:
👍

Reply:
Something I just realized too, it might be useful to generate log files. There is a logging module that might be of use. The cleaner output could go to step it, while the more verbose to a log file? 

Not sure if that is the best way to use a log file however. But it would be nice to have an auto generated file for errors and successful runs.